ceremonies/src/benchmarking.rs

```diff
@@ -0,0 +1,386 @@
+		for i in 0..9 {
+			let p = sr25519::Pair::from_entropy(&[i as u8; 32], None).0;
+			attestees.push(p.clone());
+			if i < 7 {
+				proofs.push(get_proof_of_attendance::<T>(&p.clone(), cid));
+			}
+		}
+
+		// register users
+		for i in 0..9 {
+			let p = attestees[i as usize].clone();
+			if i < 7 {
+			assert_ok!(Pallet::<T>::register_participant(RawOrigin::Signed(account_id::<T>(&(p.clone()))).into(), cid, Some(proofs[i as usize].clone())));
+			} else {
+				assert_ok!(Pallet::<T>::register_participant(RawOrigin::Signed(account_id::<T>(&(p.clone()))).into(), cid, None));
+			}
+		}
+
+		run_to_next_phase::<T>();
+		run_to_next_phase::<T>();
+
+
+		let cindex = encointer_scheduler::Pallet::<T>::current_ceremony_index();
+		let loc = Location { lat: Degree::from_num(1i32), lon: Degree::from_num(1i32) };
+		let time = correct_meetup_time::<T>(cindex, loc);
+		let mindex = 1;
+
+
+		let claims = get_all_claims::<T>(&attestees, cid, cindex, mindex, loc,time, 10);
+		assert_eq!(AttestationCount::<T>::get((cid, cindex)), 0);
+	}: _(RawOrigin::Signed(account_id::<T>(&attestor)), claims)
+	verify {
+		assert_eq!(AttestationCount::<T>::get((cid, cindex)), 1);
+	}
+
+	endorse_newcomer {
+		let cid = create_community::<T>();
+		let alice: T::AccountId = account("alice", 1, 1);
+
+		// issue some income such that nebies are allowed to register
+		assert_ok!(encointer_balances::Pallet::<T>::issue(
+			cid,
+			&alice,
+			NominalIncome::from_num(1)
+		));
+
+		let newbie = sr25519::Pair::from_entropy(&[10u8; 32], None).0;
+		assert_ok!(Pallet::<T>::register_participant(RawOrigin::Signed(account_id::<T>(&newbie.clone())).into(), cid, None));
+		let cindex = encointer_scheduler::Pallet::<T>::current_ceremony_index();
+
+		assert_eq!(<EndorseesCount<T>>::get((cid, cindex)), 0);
+		}: _(RawOrigin::Signed(alice), cid, account_id::<T>(&newbie))
+	verify {
+		assert_eq!(<EndorseesCount<T>>::get((cid, cindex)), 1);
+	}
+
+	claim_rewards {
+		frame_system::Pallet::<T>::set_block_number(frame_system::Pallet::<T>::block_number() + T::BlockNumber::from(1u64)); // this is needed to assert events
+		let cid = create_community::<T>();
+
+		let mut users: Vec<sr25519::Pair> = vec![];
+		let mut proofs: Vec<ProofOfAttendance<T::Signature, T::AccountId>> = vec![];
+		// create users and fake reputation
+		for i in 0..10 {
+			let p = sr25519::Pair::from_entropy(&[i as u8; 32], None).0;
+			users.push(p.clone());
+			if i < 8 {
+				proofs.push(get_proof_of_attendance::<T>(&p.clone(), cid));
+			}
+		}
+
+		// register users
+		for i in 0..10 {
+			let p = users[i as usize].clone();
+			if i < 8 {
+			assert_ok!(Pallet::<T>::register_participant(RawOrigin::Signed(account_id::<T>(&(p.clone()))).into(), cid, Some(proofs[i as usize].clone())));
+			} else {
+				assert_ok!(Pallet::<T>::register_participant(RawOrigin::Signed(account_id::<T>(&(p.clone()))).into(), cid, None));
+			}
+		}
+
+		run_to_next_phase::<T>();
+		run_to_next_phase::<T>();
```

Comment:
This is very redundant with the setup code from the `attest_claims` benchmark. I guess you can easily extract some functions here.